core/code/chat.js

```diff
@@ -432,6 +432,15 @@ window.chat.writeDataToHash = function(newData, storageHash, isPublicChannel, is
 //
 
 window.chat.renderText = function (text) {
+  if (text.team) {
+    var teamId = window.teamStringToId(text.team);
+    if (teamId === TEAM_NONE) teamId = TEAM_MAC;
+    var spanClass = window.TEAM_TO_CSS[teamId];
+    return $('<div>')
+      .append($('<span>', { class: spanClass, text: text.plain }))
+      .html();
+  }
+
   return $('<div>').text(text.plain).html().autoLink();
```

Comment:
Do we really need two returns here?
It should be enough to fill the spanClass conditionally and always return the same construct, yes?

Reply:
yes, except if we want to add useless `<span>` to all text parts.
or remove the jquery codes and use plain text.

atleast I don't see a better and clearer solution here.

(note: the `<div>`  tag here is removed the `.html()` call )

Reply:
And why add span into empty div? Just for coloring? That's also useless span, isn't it? 🤷

Reply:
yes, just for coloring. how do you apply colors to text parts? 😕

Reply:
I thing I was a little bit misunderstandable:
`$("<div>).text("Hello").html()` = `"Hello"`

Reply:
Exactly, through the div 🤷

You now have `<div><span class="enl">text</></>`
But it's really enough to create only `<div class="enl">text</>`

Reply:
there is no div.
`.hmtl() ` will return the innerhtlm code. not the tag itself.

Reply:
the function returns the string `<span class="enl">text</span>`, it doesn't return <div><span class="enl">text</></>

Reply:
Then why the hell create the div in the first place? 🙈

Reply:
bc it's a common pattern in iitc codebase and jquery codes. You can replace them by `.prop('outerHTML')` since jquery doesn't provide a direct method for outer html

Reply:
So you are basically saying that this
```$('<div>') 
       .append($('<span>', { class: spanClass, text: text.plain })) 
       .prop("innerHTML");
```
Is better than
```return $('<span>', { class: spanClass, text: text.plain })
       .prop("outterHTML");
```

And this
```return $('<div>').text(text.plain). prop("innerHTML").autoLink();```
Is better than this
```return text.plain.autoLink();```
?

Reply:
The reason for the code is to convert a string into html code.
like: `$("<div>", {text:"<br>"}).html()` => `&lt;br&gt;`

When using `.prop("innerHTML")` you'll leave JQuery and start using DOMElement code.
I wouldn't mix it.

Reply:
Ok. I'll let it slide for some actual html, but the later case is just nonsense.

Why surround string with div just to get that string out later?
That should be fixed to only `text.plain.autolink()`

Reply:
it's just the JQuery way to do:
```
function escapeHTML(str) {
    return str.replace(/&/g,'&amp;').replace(/</g,'&lt;').replace(/>/g,'&gt;') ;
}
```
or
```
var escape = document.createElement('textarea'); // or div 
function escapeHTML(html) {
    escape.textContent = html;
    return escape.innerHTML;
}
```
( examples taken from: https://stackoverflow.com/questions/5499078/fastest-method-to-escape-html-tags-as-html-entities )
if I recall correclty there is also an IITC helper function in utils.js. This could be used as alternative here.